### PR TITLE
Remove OpenShift-specific Dockerfiles.

### PR DESCRIPTION
**Pull request by benluddy**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,56 +1,36 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
```
Comment (timflannagan) [inline Dockerfile:1]: Do we want to wait until all the prow-based jobs, which rely on this Dockerfile and we substitute images based on known base image names, until after they all have been migrated to GH actions?

Comment (timflannagan) [reply Dockerfile:1]: Yeah, same comment w.r.t to the base.Dockerfile, which is listed as the build root Dockerfile in the o/release prow configuration. I'm working towards removing the prow tests after I get the marketplace k8s bump off my plate.

Comment (benluddy) [reply Dockerfile:1]: Definitely, let me know if I can help with that.

/hold

Comment (timflannagan) [reply Dockerfile:1]: @benluddy I have a PR open for adding the unit/verify tests to GH actions in https://github.com/operator-framework/operator-lifecycle-manager/pull/2096 and once that merges we can merge the [o/release PR](https://github.com/openshift/release/pull/18324) for removing these OCP specific Dockerfile(s).

Comment (timflannagan) [reply Dockerfile:1]: Both of those aformentioned PRs merged - I'm on the fence whether we want to just remove the OCP-specific Dockerfile(s) here (e.g. Dockerfile, base.Dockerfile) and keep the upstream.Dockerfile w.o. deflating/merging with a single Dockerfile. I think it's fine to iterate on this going forward as our upstream CI doesn't test using the same container image we attempt to successfully build.

**Comment by timflannagan** [comment]
I'd like to get another person to sign off on the changes, but this seems reasonable to me and is largely a no-op now that we're removed prow from this repository.

/lgtm
/hold

**Comment by benluddy** [comment]
> /rerun-workflow unit

https://github.com/operator-framework/operator-lifecycle-manager/issues/2092

File: Dockerfile (hunk 1 above)
Comment (benluddy) [reply Dockerfile:1]: > just remove the OCP-specific Dockerfile(s) here

That's all that is happening here -- Dockerfile and base.Dockerfile are removed, and upstream.Dockerfile is renamed to Dockerfile with no changes. The diff is just misleading.

**Pull request merged**